VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionCommentsFragment.java

```diff
@@ -159,8 +162,12 @@ public void onDestroy() {
 
     @SuppressWarnings("unused")
     public void onEventMainThread(DiscussionCommentPostedEvent event) {
-        if (null != event.getParent() && event.getParent().getIdentifier().equals(discussionResponse.getIdentifier())) {
-            discussionCommentsAdapter.insertCommentAtEnd(event.getComment());
+        if (!hasMorePages) {
+            if (null != event.getParent() && event.getParent().getIdentifier().equals(discussionResponse.getIdentifier())) {
+                ((BaseFragmentActivity)getActivity()).showInfoMessage(getString(R.string.discussion_comment_posted));
```

Comment:
The message should be shown even if there are more pages to load yet. In fact, that is the scenario where it really needs to be shown :stuck_out_tongue: If the item is displayed, then we could even dispense with it.